alacritty_terminal/src/term/mod.rs

```diff
@@ -1827,15 +1827,19 @@ impl<T: EventListener> Handler for Term<T> {
             Attr::Italic => cursor.template.flags.insert(Flags::ITALIC),
             Attr::CancelItalic => cursor.template.flags.remove(Flags::ITALIC),
             Attr::Underline => {
-                cursor.template.flags.remove(Flags::DOUBLE_UNDERLINE);
+                cursor.template.flags.remove(Flags::DOUBLE_UNDERLINE | Flags::UNDERCURL);
```

Comment:
Curious why you went for this approach. Especially when we add support for dotted/dashed underline this will likely get even "longer". So having a single flag that includes all underline flags likely makes sense for removal.

You have `all_underlines` as a function, but I don't understand why it's a function and not consistently used everywhere?